api/src/services/extension-libs.js

```diff
@@ -0,0 +1,36 @@
+const db = require('../db');
+
+const DOC_ID = 'extension-libs';
+
+const getLibsDoc = async () => {
+  try {
+    return await db.medic.get(DOC_ID, { attachments: true });
+  } catch (err) {
+    if (err.status === 404) {
+      // no doc means no configured libs
+      return;
+    }
+    throw err;
+  }
+};
+
+module.exports = {
+  isLibChange: (change) => (change && change.id) === DOC_ID,
+  getAll: async () => {
+    const doc = await getLibsDoc();
+    if (!doc || !doc._attachments) {
+      return [];
+    }
+    return Object.entries(doc._attachments).map(([ name, attachment ]) => ({ name, attachment }));
+  },
+  get: async (libName) => {
+    const doc = await getLibsDoc();
+    const attachment = doc && doc._attachments && doc._attachments[libName];
+    if (attachment) {
+      return {
+        data: attachment.data,
+        contentType: attachment.content_type
+      };
```

Comment:
At the risk of over-architecting, I wonder if we should consider building in the concept of an `extension_type` right from the beginning (the value for these would be something like `xpath_function`)?  My thought is that in the future we will probably want to add other types of extensions besides just custom xpath functions. It might make sense to include those extensions as attachments to this `extension-libs` doc too, but it would be good to be able to differentiate.

Reply:
Trying to be forwards compatible is hard... I don't think we need to change anything in code, because we can change that easily enough when we want to use the `extension_type` property. To be safe we _could_ store something in the doc, so when that change happens we can just add a new type and there's no migration. However because this is only modifying one doc it's easy enough to migrate, or just handle the "unknown" case by defaulting to `xpath_function`.

I'm reluctant to change anything at this stage because I don't know enough about what other extensions will look like...

Reply:
Fair.  Probably best to not try to blindly solve problems we don't have yet....